sycl/include/sycl/ext/intel/esimd/detail/sycl_util.hpp

```diff
@@ -74,12 +85,26 @@ struct is_sycl_accessor_with
               (is_sycl_accessor<T>::target == AccessTarget),
           std::true_type, std::false_type> {};
 
+template <typename T, accessor_mode_cap_val_t Capability>
+struct is_local_accessor_with
+    : public std::conditional_t<
+          sycl::detail::acc_properties::is_local_accessor_v<T> &&
+              accessor_mode_has_capability<local_accessor_access_mode<T>::mode,
+                                           Capability>(),
+          std::true_type, std::false_type> {};
+
+template <typename T, accessor_mode_cap_val_t Capability,
+          sycl::access::target AccessTarget>
+struct is_accessor_with
+    : public std::conditional_t<
+          detail::is_sycl_accessor_with<T, Capability, AccessTarget>::value ||
```

Comment:
It would be better to rename this from is_sycl_accessor_with to is_device_accessor_with